src/components/CartItem/CartItem.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+/* eslint-disable global-require */
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import { CartItemType } from '../../types/CartItemType';
+import './CartItem.scss';
+import { GeneralContext } from '../../helpers/GeneralContext';
+import { BASE_URL } from '../../api/api';
+
+type Props = {
+  product: CartItemType;
+};
+
+export const CartItem: React.FC<Props> = ({ product }) => {
+  const { cartList, setCartList } = useContext(GeneralContext);
+
+  const deleteItemFromCart = () => {
+    const newCartList = cartList.filter(item => {
+      return item.id !== product.id;
+    });
+
+    setCartList(newCartList);
+  };
+
+  const changeNumber = (change: number) => {
+    const changedProduct = product;
+    const index = cartList.findIndex(item => item.id === product.id);
+
+    changedProduct.quantity += change;
+
+    const newCart = [...cartList];
+
+    newCart.splice(index, 1, changedProduct);
+
+    setCartList(newCart);
+  };
+
+  const increaseNumber = () => changeNumber(1);
```

Comment:
Why don't you use changeNumber as a handler directly? 
Having that extra functions looks unnecessary.